### PR TITLE
Rewrite backend from scratch

### PR DESCRIPTION
**Pull request by WumboSpasm**
Backend changes:
- Added support for direct links
- Revamped logging system (may need further changes)
- NSFW filter is no longer hardcoded and is now togglable
- Votes are now returned as part of entry data

Frontend changes:
- Added support for direct linking as well as in-page button
- Added filter toggle for random entries as well as in-page buttons
- Added compatibility rating alongside vote
- Added error messages for when a non-existent or non-Flash entry is loaded
- Improved underlying HTML/CSS

### CODE REVIEW

File: backend/main.go [hunk 1]
```diff
@@ -1,215 +1,241 @@
...
+func addVote(id string, q string) string {
+    if !verifyUUID(id) {
+        return "UUID is not valid"
+    }
+    
+    row := flashpointDB.QueryRow(fmt.Sprintf(`
+        SELECT id
+        FROM   game
+        WHERE  id = '%v'
+    `, id))
```
Comment (LindirQuenya) [inline backend/main.go:211]: Ahem SQLi.

**Comment by LindirQuenya** [comment]
Hm, looks like all the other ones aren't sanitized right either. Please use a parameterized query.

**Comment by WumboSpasm** [comment]
What do you mean? The `verifyUUID` function is called wherever a user-defined UUID is present.

**Comment by WumboSpasm** [comment]
Ahhh I see, you can pass variables directly through the Query/Exec methods. Will fix.

File: backend/main.go [hunk 2]
```diff
@@ -1,215 +1,241 @@
...
+            fpRow := flashpointDB.QueryRow(`
+                SELECT   id, title, launchCommand, tagsStr 
+                FROM     game 
+                WHERE    launchCommand LIKE '%.swf'
+                ORDER BY random()
+                LIMIT    1
+            `)
+            fpErr := fpRow.Scan(&entry.UUID, &entry.Title, &entry.LaunchCommand, &tags)
+            if fpErr != sql.ErrNoRows && fpErr != nil {
+                log.Fatal(fpErr)
```
Comment (Dri0m) [inline backend/main.go:113]: not a fan at all, this will just dump panics into the log and return 500, but I accept it for now

**Review by Dri0m — APPROVED** [review]

**Pull request merged**